phoenix-core/src/main/java/org/apache/phoenix/execute/HashJoinPlan.java

```diff
@@ -462,7 +462,9 @@ public ServerCache execute(HashJoinPlan parent) throws SQLException {
                 }
 
                 Object result = expectSingleRow ? (values.isEmpty() ? null : values.get(0)) : PArrayDataType.instantiatePhoenixArray(baseType, values.toArray());
-                parent.getContext().setSubqueryResult(select, result);
+                if (result != null) {
```

Comment:
Nit: Can you switch the if to return null in the then clause ?

Reply:
thanks for reviewing
for the org.apache.phoenix.execute.HashJoinPlan.WhereClauseSubPlan#execute, it always return null;
you mean change here like this?
                if (result == null) {
                    return null;
                }
                parent.getContext().setSubqueryResult(select, result);
                return null;
here i changed just judge to avoid put a null value to subqueryResults as a subquery result

Reply:
Forget it, your code is fine.
We always have to return null anyway.